cardano-api/src/Cardano/Api/LedgerState.hs

```diff
@@ -1521,16 +1533,21 @@ currentEpochEligibleLeadershipSlots sbe sGen eInfo pParams ptclState
 
   -- We need the "set" stake distribution (distribution of the previous epoch)
   -- in order to calculate the leadership schedule of the current epoch.
-  let setSnapshotPoolDistr :: Map.Map (ShelleyAPI.KeyHash 'ShelleyAPI.StakePool Shelley.StandardCrypto) (Ledger.IndividualPoolStake Shelley.StandardCrypto)
+  let setSnapshotPoolDistr :: Map (SL.KeyHash 'SL.StakePool Shelley.StandardCrypto) (SL.IndividualPoolStake Shelley.StandardCrypto)
       setSnapshotPoolDistr = ShelleyAPI.unPoolDistr . ShelleyAPI.calculatePoolDistr
                                 . ShelleyAPI._pstakeSet . obtainIsStandardCrypto sbe
                                 $ ShelleyAPI.esSnapshots cEstate
 
-  relativeStake :: Rational <- maybe (Left $ LeaderErrStakePoolHasNoStake poolid)
-                         (Right . ShelleyAPI.individualPoolStake)
-                         (Map.lookup poolHash setSnapshotPoolDistr)
+  let slotRangeOfInterest = Set.filter
+        (not . Ledger.isOverlaySlot firstSlotOfEpoch (getField @"_d" (toLedgerPParams sbe pParams)))
```

Comment:
I think we (eventually?) won't have `d` at all for Babbage/Praos, right? Will this require an update then?

Reply:
There is no longer an overlay schedule in Praos, so this line could be simplified to just return the set of slots in the given epoch.

(As for removing `d`, it is not in the Babbage era protocol parameters, but the `HasField` instance still [exists](https://github.com/input-output-hk/cardano-ledger/blob/2505b7f103c78ee5a230b3e302e31d9607fffbd5/eras/babbage/impl/src/Cardano/Ledger/Babbage/PParams.hs#L460-L464), sadly, since `d` has its tendrils all throughout the reward calculation and we chose not to redo all that code.)

Reply:
Since we need to keep it for `TPraos` anyway, I'm happy to keep this code for the moment and be guided by the type errors if/when the `d` parameters get deleted for Babbage onwards.